### PR TITLE
Update main.py (added csv export)

### PR DESCRIPTION
**Pull request by CaltropHungerton**
added a function that exports the accuracy, speed, and size results as csv files.

### CODE REVIEW

**Comment by cyrilou242** [comment]
Nice thanks! Adding a few comments

File: main.py [hunk 1]
```diff
@@ -10,6 +11,12 @@
 from compressors.zstd_compressor import ZstdCompressor
 from data import load_20news, load_ohsumed_single_23, load_reuters, load_kinnews_kirnews
 
+def write_csv(filename, data, headers):
```
Comment (cyrilou242) [inline main.py]: should this be moved to another file? 

(I don't do much python so I don't have a strong opinion)

File: main.py [hunk 2]
```diff
@@ -130,13 +137,23 @@ def run_experiment(dataset, compressor, top_k_accuracy, compressors_per_class, s
                     size_results.append(size_result)
 
     # todo add save in proper tabular format
```
Comment (cyrilou242) [inline main.py]: you can remove this comment :)

File: main.py [hunk 3]
```diff
@@ -130,13 +137,23 @@ def run_experiment(dataset, compressor, top_k_accuracy, compressors_per_class, s
...
+
+    #so basically i create list comprehension of all keys in first dict
```
Comment (cyrilou242) [inline main.py]: let's remove this comment
if you want to make things clearer, I suggest: 

```
accuracy_headers = [x for x in results[0].keys()]
write_csv(os.path.join(os.getcwd(), 'accuracy_results.csv'), results, accuracy_headers)

...
speed_result_headers = [x for x in speed_results[0].keys()]
write_csv(os.path.join(os.getcwd(), 'speed_results.csv'), speed_results, speed_result_headers)

other option is to encapsulate this in the `write_csv` function

File: main.py [hunk 4]
```diff
@@ -130,13 +137,23 @@ def run_experiment(dataset, compressor, top_k_accuracy, compressors_per_class, s
...
+    
+    write_csv(os.path.join(os.getcwd(), 'accuracy_results.csv'), results, [x for x in results[0].keys()])
+    write_csv(os.path.join(os.getcwd(), 'speed_results.csv'), speed_results, [x for x in speed_results[0].keys()])
+    write_csv(os.path.join(os.getcwd(), 'size_results.csv'), size_results, [x for x in size_results[0].keys()])
+
+"""
```
Comment (cyrilou242) [inline main.py]: why is this commented? do we want to keep it or to remove it? 
I think it can be handy even if we want CSVs

**Review by cyrilou242 — CHANGES_REQUESTED** [review]

File: main.py (hunk 1 above)
Comment (CaltropHungerton) [reply main.py]: It probably should (weird to have straggly function definition in main). What other file would be best to move it to?

File: main.py (hunk 4 above)
Comment (CaltropHungerton) [reply main.py]: Ok, I'll uncomment it. I wasn't sure whether you wanted to keep it or not.

File: main.py (hunk 3 above)
Comment (CaltropHungerton) [reply main.py]: I encapsulated all of it into the function. It only takes the two arguments now, and handles the path stuff for the first argument within the function.

File: main.py (hunk 1 above)
Comment (cyrilou242) [reply main.py]: let's keep it in the main now that there's a tight coupling between the csv write format and the table print format.
the main is <150 lines which is reasonable enough

**Comment by cyrilou242** [comment]
LGTM thanks! 

Could you add a mention in the readme that the results are written to `accuracy_results.csv`, `speed_results.csv` and `size_results.csv`?

**Comment by CaltropHungerton** [comment]
> LGTM thanks!
> 
> Could you add a mention in the readme that the results are written to `accuracy_results.csv`, `speed_results.csv` and `size_results.csv`?

I also got rid of the TODO bullet point. I think there are a couple of other bullet points that we might be able to get rid of (add ohsumed dataset, has pytorch already been removed?) but you would know better than I would.

**Comment by cyrilou242** [comment]
> > LGTM thanks!
> > Could you add a mention in the readme that the results are written to `accuracy_results.csv`, `speed_results.csv` and `size_results.csv`?
> 
> I also got rid of the TODO bullet point. I think there are a couple of other bullet points that we might be able to get rid of (add ohsumed dataset, has pytorch already been removed?) but you would know better than I would.

Thanks! 
I'll clean up the todo.

**Review by cyrilou242 — APPROVED** [review]

**Pull request merged**